src/metal/scheduler.ts

```diff
@@ -150,20 +150,28 @@ export class ElementScheduler extends BaseScheduler implements ElementSchedulerI
   }
   applyQueue(frame: Frame) {
     for (let i = 0; i < this.queue.items.length; i++) {
-      let { callback, el, id } = this.queue.items[i];
-      let bcr = el.getBoundingClientRect();
+      let { callback, el, id, bcr } = this.queue.items[i];
+
+      if (W.isDirty || !bcr) {
+        bcr = el.getBoundingClientRect();
+      }
 
       callback(frame, id, bcr);
     }
+
+    W.isDirty = false;
```

Comment:
the WindowProxy should likely default its `isDirty` value: https://github.com/lynchbomb/spaniel/blob/ee920fdb403dff316230a17bfc3c00bc73e5bf86/src/metal/window-proxy.ts#L37

* That way we don't leak `undefined` into the system
* We don't change the shape of `W` (the WindowProxy)
* TypeScript typings can provide value